### ISSUE TITLE
TypeScript SDK: Bug preventing MSBuild reporting SDK version fallback following TS errors

### ISSUE THREAD

**Issue by mattbrooks2010** [Microsoft/TypeScript#18385]
**TypeScript SDK Version:**
2.5

**Code**
In `TypeScript.Tasks.VsTsc.FindToolPath()`:

```csharp
if (string.IsNullOrEmpty(this.ToolsVersion))
{
    base.Log.LogWarning(Strings.ToolsVersionWarning, new object[] { this.ToolsVersion });
}
```

```csharp
if (string.IsNullOrEmpty(this.ToolsVersion))
{
    base.Log.LogWarning(Strings.FallbackVersionWarning, new object[] { current.ToString() });
}
```

**Expected behavior:**
When a Visual Studio / MSBuild project file specifies a specific `<TypeScriptToolsVersion>`, `Microsoft.TypeScript.targets` should warn that it has instead actually selected the latest version of the SDK present on the host machine, perhaps using one or both of the following existing resource messages:

    ToolsVersionWarning=Couldn't locate the compiler version ({0}) specified in the project file.
    FallbackVersionWarning=Using compiler version ({0}), if this is incorrect change the <TypeScriptToolsVersion> element in your project file.

When a specific `<TypeScriptToolsVersion>` is specified in a project file, the tooling should **always** warn when that version is not found and a different version is used instead. It should do this regardless of whether there are other tooling / compiler errors.

**Actual behavior:**
When the build fails with an error, the warning is never displayed.

This causes problems in builds when the latest SDK / compiler has introduced breaking changes. The problems are initially hard to diagnose, because there is no messages in the build log indicating that a new SDK / compiler version has been used, or what SDK / compiler version is expected by the project.

**Additional info:**
Please note that the code in the method above may need to also consider a `<TypeScriptToolsVersion>` of `Latest` OR the Visual Studio `TypeScript Build` properties window should be modified to write an empty string or remove the property completely.

**Repro:**
1. Install TypeScript SDK 2.5.
1. Create a blank Visual Studio project and add a TypeScript file to the build.
1. Modify the TypeScript file to trigger a build error, e.g. `const x = 1; x = 2;`
1. Change `<TypeScriptToolsVersion>` to `0.0` where `0.0` is not present at `C:\Program Files (x86)\Microsoft SDKs\TypeScript`. 
1. Change MSBuild output to `Diagnostic`.
1. Rebuild solution.
1. Observe zero warnings.
1. Observe that version 2.5 of the compiler was selected by searching for `tsc.exe` in the MSBuild output.

**Comment by mattbrooks2010** [comment]
Repro attached — [WebApplication1.zip](https://github.com/Microsoft/TypeScript/files/1292668/WebApplication1.zip).

**Comment by minestarks** [comment]
Thanks @mattbrooks2010 for this. 
This is fixed in the nightly nuget, will likely be included in the next update to 2.9 if there is one.
